pkg/trace/parser.go

```diff
@@ -101,19 +102,26 @@ func (p *AzureTraceParser) extractFunctions(
 		dirigentMetadataByHashFunction = createDirigentMetadataMap(dirigentMetadata)
 	}
 
+	gen := rand.New(rand.NewSource(time.Now().UnixNano()))
+
 	for i := 0; i < len(*invocations); i++ {
 		invocationStats := (*invocations)[i]
 
 		function := &common.Function{
-			Name: fmt.Sprintf("%s-%d-%d", common.FunctionNamePrefix, i, p.functionNameGenerator.Uint64()),
+			Name: fmt.Sprintf("%s-%d", "t"+invocationStats.HashFunction[0:18], rand.Uint64()),
 
 			InvocationStats: &invocationStats,
 			RuntimeStats:    runtimeByHashFunction[invocationStats.HashFunction],
 			MemoryStats:     memoryByHashFunction[invocationStats.HashFunction],
+
+			ColdStartBusyLoopMs: generator.StartupLoopConvertMemoryToRuntimeMs(generator.GenerateMemorySpec(gen, gen.Float64(), memoryByHashFunction[invocationStats.HashFunction])),
 		}
 
 		if dirigentMetadata != nil {
 			function.DirigentMetadata = dirigentMetadataByHashFunction[invocationStats.HashFunction]
+		} else if strings.Contains(strings.ToLower(platform), "knative") {
```

Comment:
From what I understand the purpose of this feature is to run knative traces in Dirigent? Is that correct? I am not a big fan of this approach. I think each trace should be for one system and that's it